src/plugins/data_views/common/data_views/data_view.ts

```diff
@@ -667,6 +678,15 @@ export class DataView implements DataViewBase {
     delete this.fieldFormatMap[fieldName];
   };
 
+  private getMappedFieldNames() {
+    return this.fields.getAll().reduce<string[]>((acc, dataViewField) => {
+      if (dataViewField.isMapped) {
+        return [...acc, dataViewField.name];
```

Comment:
use `acc.push()` instead of `[...acc, new value]` - the code unfortunately does exactly what it looks like it does, it creates a new array, copying the old array, and adding an additional value. This is fine for smaller arrays but once you get into the thousands of elements then it can become slow.

Reply:
766682de3c7